infra/src/main/java/com/redhat/service/bridge/infra/auth/IdentityResolver.java

```diff
@@ -3,5 +3,8 @@
 import org.eclipse.microprofile.jwt.JsonWebToken;
 
 public interface IdentityResolver {
-    String resolve(JsonWebToken jwt);
+
+    String getCustomerIdFromUserToken(JsonWebToken jwt);
```

Comment:
@r00ta My only comment is if we actually need to separate these into separate methods rather than simply preserving the `resolve` method and then internally within the `IdentityResolverImpl`, working out if the user is trying to auth with a Serivce Account or a User Account.

As part of https://issues.redhat.com/browse/MGDOBR-423 I was going to include allowing a user to use a Service Account to auth to our main API, so we'd kinda get that for "free" if we encapsulate the logic within a single `resolve` method.

WDYT?

Reply:
> I think it depends on the behaviour we expect from our api. My understanding was that
> 
> 1. the manager api can be called only by users (no service accounts)

We can support Service Accounts on our public API. It's something we have requested from Managed Kafka to support our Fleet Manager invoking their API with a Service Account, so I don't see any reason for us not to also support it.

> 2. the ingress can be called only by service accounts (no users).
> 

The use-case I could see here is allowing the user to send a test event to their bridge instance from console.redhat.com? We haven't really started to think about tutorials/quick-starts yet, but my expectation is that a lot of them will be driven via the console.redhat.com UI.

> In order to achieve this the api should reject user tokens if it expects a service account token and vice-versa.
> 
> 1. If we want to accept both in the api, then we can go for the single `resolve` method
> 2. if we want to keep them separated, I'd stay with `getCustomerIdFromUserToken` and `getCustomerIdFromServiceAccountToken` so that we can raise an exception if the submitted token should not be accepted.
> 
> wdyt?

Reply:
@r00ta Yes, I think we should.

Reply:
done